src/crimson/os/seastore/transaction_manager.h

```diff
@@ -955,23 +926,23 @@ class TransactionManager : public ExtentCallbackInterface {
   shard_stats_t& shard_stats;
 
   template <typename T>
-  std::variant<LBAMappingRef, get_child_ifut<T>>
+  std::variant<LBAMapping, get_child_ifut<T>>
   get_extent_if_linked(
     Transaction &t,
-    LBAMappingRef pin)
+    LBAMapping pin)
   {
-    ceph_assert(pin->is_parent_viewable());
+    ceph_assert(pin.is_valid());
     // checking the lba child must be atomic with creating
     // and linking the absent child
-    auto v = pin->get_logical_extent(t);
+    auto v = pin.get_logical_extent(t);
     if (v.has_child()) {
       return v.get_child_fut(
       ).si_then([pin=std::move(pin)](auto extent) {
 #ifndef NDEBUG
         auto lextent = extent->template cast<LogicalChildNode>();
-        auto pin_laddr = pin->get_key();
-        if (pin->is_indirect()) {
-          pin_laddr = pin->get_intermediate_base();
+        auto pin_laddr = pin.get_key();
+        if (pin.is_indirect()) {
+          pin_laddr = pin.get_intermediate_base();
```

Comment:
It should be safe to reduce the above 3 lines into `auto intermediate_base = pin.get_intermediate_base()`, and drop the internal `assert(is_indirect());`.